x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elser/ElserMlNodeService.java

```diff
@@ -154,11 +162,33 @@ public void start(Model model, ActionListener<Boolean> listener) {
         startRequest.setThreadsPerAllocation(serviceSettings.getNumThreads());
         startRequest.setWaitForState(STARTED);
 
-        client.execute(
-            StartTrainedModelDeploymentAction.INSTANCE,
-            startRequest,
-            listener.delegateFailureAndWrap((l, r) -> l.onResponse(Boolean.TRUE))
-        );
+        client.execute(StartTrainedModelDeploymentAction.INSTANCE, startRequest, elserNotDownloadedListener(model, listener));
+    }
+
+    private static ActionListener<CreateTrainedModelAssignmentAction.Response> elserNotDownloadedListener(
+        Model model,
+        ActionListener<Boolean> listener
+    ) {
+        return new ActionListener<>() {
+            @Override
+            public void onResponse(CreateTrainedModelAssignmentAction.Response response) {
+                listener.onResponse(Boolean.TRUE);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                if (e instanceof ResourceNotFoundException) {
```

Comment:
`StartTrainedModelDeploymentAction` is a master node action which means the request will most likely be executed on a different node, in which case root cause may be wrapped in another exception. 

```suggestion
                if (ExceptionsHelper.unwrapCause(e) instanceof ResourceNotFoundException) {
```